content/blog/accessing-hpe-data-fabric-s3-storage-from-spring-boot-s3-micro-service-deployed-in-k3s-cluster.md

```diff
@@ -0,0 +1,155 @@
+
+7.      Execute below docker commands to build docker image and push it to docker hub. 
+
+Note: Alternatively , we can use podman instead of docker to create images. More information on podman can be obtained from [here](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli) [](https://developers.redhat.com/blog/2020/11/19/transitioning-from-docker-to-podman#transition_to_the_podman_cli)
+
+```bash
+docker build -f Dockerfile -t <Dockerhub user id>/df-s3-springboot-k3s-demo:latest .
+docker image ls
+docker login -u <Dockerhub user id>
+> enter password:  <Dockerhub password>
+docker push <Dockerhub user id>/df-s3-springboot-k3s-demo:latest
+
+```
+
+8.      Create below df-s3-springboot-k3s-demo.yaml file for deploying the executable in K3s cluster. Sample yaml file is given in project directory. Please replace <dockerhub userid> with valid id.
+
+```Yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: df-s3-springboot-k3s-demo-service
+spec:
+  selector:
+    app: df-s3-springboot-k3s-demo
+  ports:
+  - protocol: TCP
+    name: df-s3-springboot-k3s-demo
+    port: 8000
+    targetPort: 8000
+  type: LoadBalancer
+
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: df-s3-springboot-k3s-demo
+spec:
+  selector:
+    matchLabels:
+      app: df-s3-springboot-k3s-demo
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: df-s3-springboot-k3s-demo
+    spec:
+      containers:
+      - name: df-s3-springboot-k3s-demo
+        image: <Dockerhub userid>/df-s3-springboot-k3s-demo:latest
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 8000
+```
+
+9.      Before deploying it in Kubernetes cluster, validate the docker or podman image by running the image. 
+
+```bash
+docker run -p 8000:8000 <Dockerhub userid>/df-s3-springboot-k3s-demo
+```
+
+```UI
+http://localhost:8000/swagger-ui.hmtl
+```
+
+**Deploying in K3s cluster**
+
+1.      Log into the control plane node of the K3s cluster. Create/Copy the df-s3-springboot-k3s-demo.yaml to the node.
+
+2.      Execute “kubectl apply -f df-s3-springboot-k3s-demo.yaml”. If required, you can specify the namespace option.
+
+3.      Check the pod creation status by using the command “kubectl get pods -l app=df-s3-springboot-k3s-demo -o wide”
+
+4.      Verify if the services are properly deployed by using the command “kubectl get service df-s3-springboot-k3s-demo-service”
+
+**Accessing the Swagger UI from the pod**
+
+1.      Connect to the pod http://pod-ip:8000/swagger-ui.html
+
+2.      Verify the services exposed in the Swagger-UI
+
+![Swagger UI](/img/swagger-ui.png)
+
+In this blog , you learnt how data can be processed from edge to persistence store by creating data pipeline using different technology stack. The mentioned data pipeline is not limited to current use case but can be used in diversified microservice use cases. 
```

Comment:
Nice! You've got the idea. Just some grammatical edits here and it will be perfect:

![image](https://user-images.githubusercontent.com/77496261/134182356-f2142a87-b343-46d3-9f44-e10e43d384d2.png)


So it would read:
In this blog post, you learned how data can be processed from edge to a persistence store by creating a data pipeline using a different technology stack. The aforementioned  data pipeline is not limited to this current use case but can be used in many diversified microservice use cases.

Reply:
done.